tls/s2n_connection.h

```diff
@@ -85,6 +101,11 @@ struct s2n_connection {
     uint8_t actual_protocol_version;
     uint8_t actual_protocol_version_established;
 
+    /* Certificate Authentication and Verification Parameters */
+    s2n_cert_auth_type client_cert_auth_type;
```

Comment:
I guess my comment was also implicitly asking for us to change our checks in the handshake to look at multiple conditions (auth type and mode) in addition to renaming. If that is too big of a refactor, then I don't want to block this PR on it.